src/source/image_source.js

```diff
@@ -111,6 +128,42 @@ class ImageSource extends Evented implements Source {
             }
         });
     }
+    
+    /**
+     * Updates the image URL and, optionally, the coordinates. To avoid having the image flash after changing,
+     * set the `raster-fade-duration` paint property on the raster layer to 0.
+     *
+     * @param {Object} options
+     * @param {string} [options.url] Required image URL.
+     * @param {Array<Array<number>>} [options.coordinates] Four geographical coordinates,
+     *   represented as arrays of longitude and latitude numbers, which define the corners of the image.
+     *   The coordinates start at the top left corner of the image and proceed in clockwise order.
+     *   They do not have to represent a rectangle.
+     * @returns {ImageSource} this
+     */
+    updateImage(options) {
+        if (!this.image || !options.url) {
+            return this;
+        }
+
+        var updateCoords = Boolean(options.coordinates);
+
+        getImage(this.map._transformRequest(options.url, ResourceType.Image), (err, image) => {
```

Comment:
There's a lot of repetition of code in this. We could DRY it up quite a bit by using the existing `load` and `_finishLoading` functions which has the added bonus of firing the appropriate events and updating `this.url`, etc. You have to change `_finishLoading` so it can accept the coordinates as a parameter and it looks like you'd have to change `load` to accept a callback so you can pass in the necessary functionality like `this.texture = null` but otherwise it seems to work pretty well.